experimental/Schemes/Schemes.jl

```diff
@@ -0,0 +1,2587 @@
+import AbstractAlgebra.Ring, Oscar.AlgHom, Oscar.compose, AbstractAlgebra.Generic.Frac
+import Base: ∘
+import Oscar: base_ring
+import AbstractAlgebra: FPModule, FPModuleElem
+import Base.copy
+import AbstractAlgebra.Generic.MatSpaceElem
+
+include( "./Multiindices.jl" )
+using Oscar.Multiindices
+
+include( "./Misc.jl" )
+using Oscar.Misc
+
+export AffineScheme, Spec, SpecPrincipalOpen
+export base_ring, ambient_ring, defining_ideal, pullback, pullback_from_parent, pullback_from_root, inclusion_in_parent, inclusion_in_root, set_name!, inverted_element, identity_morphism, denoms, inverses
+export affine_space, localize, subscheme
+
+export AffSchMorphism
+export domain, codomain, pullback
+
+export Glueing
+export first_patch, second_patch, overlap, first_inclusion, second_inclusion, glueing_isomorphism
+
+export Covering
+export patches, glueings, _add_patch!
+
+export Refinement
+
+export CoveredScheme
+export coverings
+
+export CovSchMorphism
+
+export AbstractCoherentSheaf
+
+export IdealSheaf
+export parent, covering, ideals
+
+export CoherentSheaf
+export parent, covering, modules
+
+export LineBundle, VectorBundle
+export parent, covering, transitions, rank, OO, direct_sum
+
+export VectorBundleSection, LineBundleSection
+export tautological_sections
+
+export projectivize
+export cartesian_product
+
+@Markdown.doc """
+    abstract type Scheme{ S <: Ring }
+
+The type parameter S is the type of the base ring k over which the Scheme is defined.
```

Comment:
I think we agreed overall that argument names are in backticks, as it is code; that's also what Julia does in general.
```suggestion
The type parameter `S` is the type of the base ring `k` over which the scheme is defined.
```